### PR TITLE
stableToken checks exchangeIdentifier exist in the registry on initialization

### PR DESCRIPTION
**Pull request by martinvol**
### Description

Checks that exchangeIdentifier is valid on initialization.

### Other changes

- 

### Tested

Added unit tests.

### Related issues

- 

### Backwards compatibility

-

### CODE REVIEW

**Comment by martinvol** [comment]
closes https://github.com/celo-org/celo-labs/issues/792

**Review by AlexBHarley — COMMENTED** [review]
Looks good to me but if someone with more SC experience can give it a once over that'd be nice.

File: packages/protocol/contracts/stability/StableToken.sol [hunk 1]
```diff
@@ -139,7 +139,10 @@ contract StableToken is
       _mint(initialBalanceAddresses[i], initialBalanceValues[i]);
     }
     setRegistry(registryAddress);
+
     exchangeRegistryId = keccak256(abi.encodePacked(exchangeIdentifier));
+    // fail in case exchangeRegistryId can't be looked up in the registry
+    registry.getAddressForOrDie(getExchangeRegistryId());
```
Comment (yorhodes) [inline packages/protocol/contracts/stability/StableToken.sol:145]:
```suggestion
    registry.getAddressForOrDie(exchangeRegistryId);
```

**Review by yorhodes — COMMENTED** [review]
I'm not sure we should make this change
It requires a much stronger assumption about the ordering of execution in the proposal
See discussion in [slack thread](https://celo-org.slack.com/archives/CJWE2KURZ/p1612866060287100)

**Comment by martinvol** [comment]
We agreed we wont fix this.

**Pull request closed without merging**